### ISSUE TITLE
How to configure multiple logger like standard logging.config.dictConfig

### ISSUE THREAD

**Issue by Gideon-koutian** [Delgan/loguru#25]
hi,I read through the readme and api reference but didnot find out how to get multiple logger instances.
here is my configure of logging loggers:
![image](https://user-images.githubusercontent.com/42484274/50462824-cc747380-09c2-11e9-89f7-999cb7cc0a7b.png)

in my project,i rely on getLogger("xxx") to process the log instead of getLogger() to distribute the log. And I don't want some of the logs to be passed to multiple loggers, so I added `propagate: no` to each logger.
in loguru's api reference，I only found the `filter Parameters in logger.add` to handle multiple loggers.
but mine multiple loggers are stored in the same log, if i used `filter=“xxx.log”`,log record  will send to multiple loggers,so how to deal with my scene

**Comment by Gideon-koutian** [comment]
first I changed my configuration to this
```python
logger.add(
    handlers.RotatingFileHandler(filename="log/jetty_stdout.log", mode='a', backupCount=9, maxBytes=eval('1024*1024'),
                                 encoding="utf-8", delay=True), level=logging.INFO, format=_format, enqueue=True,
    backtrace=True, filter=lambda record: True if "LOG" in record["extra"] else False)
logger.add(
    handlers.RotatingFileHandler(filename="log/scheduler_stdout.log", mode='a', backupCount=9,
                                 maxBytes=eval('1024*1024'),
                                 encoding="utf-8", delay=True), level=logging.INFO, format=_format, enqueue=True,
    backtrace=True, filter=lambda record: True if "SCHEDULER" in record["extra"] else False)
logger.add(
    handlers.TimedRotatingFileHandler(filename="log/error.log", when="midnight", interval=1, backupCount=6, utc=True,
                                      delay=False, encoding="utf-8"), level=logging.ERROR, format=_format, enqueue=True,
    backtrace=True, filter=lambda record: True if "LOG" in record["extra"] else False)
logger.add(
    handlers.TimedRotatingFileHandler(filename="log/api_jetty_stdout.log", when="midnight", interval=1, backupCount=6,
                                      utc=True, delay=False, encoding="utf-8"), level=logging.INFO, format=_format,
    enqueue=True, backtrace=True, filter=lambda record: True if "API" in record["extra"] else False)

````

Then, I replaced my original logging configuration code with this configuration.
But I found loguru missed supports the delay parameter in logging.Hander, fortunately, no effect
```python
logger.configure(**{
    "handlers": [
        {"sink": sys.stdout, "format": _format, "backtrace": True, "catch": True,
         "filter": lambda record: False if "SCHEDULER" in record["extra"] else True},
        {"sink": LOGGING_JERTY, "level": logging.INFO, "format": _format, "backtrace": True,
         "catch": True, "rotation": "10 MB", "retention": 9, "compression": "gz",
         "filter": lambda record: False if "SCHEDULER" in record["extra"] else True},
        {"sink": LOGGING_ERROR, "level": logging.ERROR, "format": _format, "backtrace": True,
         "catch": True, "rotation": "w0", "retention": 6, "compression": "gz",
         "filter": lambda record: False if "SCHEDULER" in record["extra"] else True},
        {"sink": LOGGING_SCHEDULER, "level": logging.INFO, "format": _format, "backtrace": True,
         "catch": True, "rotation": "10 MB", "retention": 1, "compression": "gz",
         "filter": lambda record: True if "SCHEDULER" in record["extra"] else False}
    ]
})
```
After that, I will try to change this code to a configuration file.😁
and I hope to have more interaction with the Hander of the standard library.

**Comment by Delgan** [comment]
I hope this is not too much burden compared to the standard logging library. :grimacing:

>  But I found loguru missed supports the delay parameter in logging.Hander, fortunately, no effect

Do you mean there is a problem with `delay` parameter of Loguru's sink? The code `logger.add("file.log", delay=True)` should work if the sink (`"file.log"` here) is a `str` or a `Path`. It has no effect if the sink is of another type.

 > I hope to have more interaction with the Hander of the standard library.

What would you suggest to Loguru?

**Comment by Delgan** [comment]
@Gideon-koutian Are you satisfied with your final configuration? Anything to add to this issue? 🙂

**Comment by Delgan** [comment]
Feel free to re-open the issue if you have others questions. 👍

**Comment by D3f0** [comment]
I'd like to allow to define/override logging settings in a separate file (a TOML file), that  provides a set of dictionaries. I see that some sink settings expect to receive sys.stderr, and reverences that cannot be "serialized". Any suggestions?

**Comment by Delgan** [comment]
Hi @D3f0. 🙂 

The problem you describe is one of the reasons I did not implement any method to load a configuration from a file.

Honestly, I don't what would be the best solution. The standard logging library partly solves it by specifying a special syntax: [_Access to external objects_](https://docs.python.org/3/library/logging.config.html#access-to-external-objects).
So, you could state that `'ext://sys.stderr'` stands for `sys.stderr`, and then tranform the TOML dict in your Python script. Depending on the dynamicity you are looking for, you could simply use pattern matching like `{"ext:://sys.stderr": sys.stderr}` or implement a proper resolver as done by the standard library: [`cpython/logging/config.py`](https://github.com/python/cpython/blob/master/Lib/logging/config.py#L375).

But when it comes to parametrizing handlers with functions, it's even more complicated. If you wish to support this too, I guess you need to define a set of pre-wrote functions in your Python script that you can parametrize in the TOML file by using their identifier.

Basically, the simplest solution would look like this I think:

```python
resolver = {
    "ext://stderr": sys.stderr,
    "ext://stdout": sys.stdout,
    "ext://database": lambda msg: db.update(msg.record),
}

for handler in toml_config["handlers"]:
     for key, value in handler.items():
         handler[key] = resolver.get(value, value)

loguru.config(toml_config)
```

I suppose you probably already have thought of this solution. I don't have any better idea for now, but I'm interested to know the solution you will choose.

**Comment by durgeksh** [comment]
I want to have two loggers.
1. Only display to console
2. Display to console and save in file

How this can be achieved?
Thank you for wonderful library.

**Comment by Delgan** [comment]
For people interest in file configuration for logging, please take a look at the `loguru-config` library: https://pypi.org/project/loguru-config

This is a great package that handles all use case of Loguru configuration.

---

@durgeksh I guess I answered you on a different ticket. Please open a new ticket if you need more help. :+1:

**Comment by durgeksh** [comment]
Thank you so much for your help. Initially, I used bind() with filter but it clubs output into single file.
When I tried to route the output of other logger to loguru, it is displaying so many internal logs. Can we filter the standard logs which need to be sent to the loguru.
My code for same is as below.

```
def get_logger():
    global _logger
    if _logger is None:
        class InterceptHandler(logging.Handler):
            def emit(self, record):
                # Get corresponding Loguru level if it exists.
                try:
                    level = logger.level(record.levelname).name
                except ValueError:
                    level = record.levelno

                # Find caller from where originated the logged message.
                frame, depth = sys._getframe(6), 6
                while frame and frame.f_code.co_filename == logging.__file__:
                    frame = frame.f_back
                    depth += 1

                logger.opt(depth=depth, exception=record.exc_info).log("INFO", record.getMessage())

        logging.basicConfig(handlers=[InterceptHandler()], level="INFO", force=True)
        _logger = lg.get_app_logger(".", LOG_FILE_PATH)

    return _logger
```
